csvutil/csv_test.go

```diff
@@ -391,7 +394,46 @@ func TestCSVWriterStruct(t *testing.T) {
 		t.Errorf("error closing table: %+v\n", err)
 	}
 
-	err = diff("testdata/simple.csv", fname)
+	err = diff("testdata/write-results.csv", fname)
+	if err != nil {
+		t.Errorf("files differ: %+v\n", err)
+	}
+}
+
+func TestCSVWriterArgsSlice(t *testing.T) {
+	fname := "testdata/out-args-slice.csv"
+	tbl, err := csvutil.Create(fname)
+	if err != nil {
+		t.Errorf("could not create %s: %+v\n", fname, err)
+	}
+	defer tbl.Close()
+	tbl.Writer.Comma = ','
+
+	err = tbl.WriteHeader("## more complicated slices: [][]int{}, [][]string{}, []string{}, float64\n")
+	if err != nil {
+		t.Errorf("error writing header: %+v\n", err)
+	}
+
+	for i := 0; i < 10; i++ {
+		var (
+			lii = [][]int{{1, 2, 3}, {2, 3, 4}, {7, 8, 15}}
+			lss = [][]string{{"foo", "bar", "baz"}, {"abc", "def", "ghi"}, {"qwerty"}}
+			ls  = []string{"abc", "def", "ghi"}
+			f   = float64(i)
+		)
+		err = tbl.WriteRow(lii, lss, ls, f)
+		if err != nil {
+			t.Errorf("error writing row %d: %+v\n", i, err)
+			break
+		}
+	}
+
+	err = tbl.Close()
+	if err != nil {
+		t.Errorf("error closing table: %+v\n", err)
+	}
+
+	err = diff("testdata/write-results-slice.csv", fname)
 	if err != nil {
 		t.Errorf("files differ: %+v\n", err)
```

Comment:
```suggestion
		t.Fatalf("files differ: %+v", err)
```
(yeah, this is probably one of my first packages for Go-HEP, it probably shows a bit: I didn't get all the idioms correctly)